SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestMultipleOwnership/SpatialTestMultipleOwnership.cpp

```diff
@@ -0,0 +1,213 @@
+
+		FinishStep();
+	});
+
+	// Client 1 sends an RPC from both MultipleOwnershipPawns
+	AddStepFromDefinition(ClientSendRPCStepDefinition, FWorkerDefinition::Client(1));
+
+	// All workers check that both MultipleOwnershipPawns have received 0 RPCs, therefore that the RPCs were ignored since the
+	// MultipleOwnershipPawns had no owner.
+	AddStep(
+		TEXT("SpatialTestMultipleOwnershipAllWorkersCheckCorrectRPCs"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+		[this](float DeltaTime) {
+			if (MultipleOwnershipPawns[0]->ReceivedRPCs == 0 && MultipleOwnershipPawns[1]->ReceivedRPCs == 0)
+			{
+				FinishStep();
+			}
+		},
+		10.0f);
+
+	// The server makes Client 1's PlayerController possess the first MultipleOwnershipPawn.
+	AddStep(TEXT("SpatialTestMultipleOwnershipPossessPawn1"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		APlayerController* PlayerController =
+			Cast<APlayerController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+		PlayerController->Possess(MultipleOwnershipPawns[0]);
+
+		FinishStep();
+	});
+
+	// Client 1 sends an RPC from both MultipleOwnershipPawns
+	AddStepFromDefinition(ClientSendRPCStepDefinition, FWorkerDefinition::Client(1));
+
+	auto PossessesCorrectPawns2 = [this]() {
+		APlayerController* PlayerController =
+			Cast<APlayerController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+		return PlayerController == nullptr || PlayerController->GetPawn() == MultipleOwnershipPawns[0];
+	};
+	// All workers check that the RPC sent from the first MultipleOwnershipPawn was correctly received, whilst the one sent from the second
+	// MultipleOwnershipPawn was ignored.
+	AddStep(
+		TEXT("SpatialTestMultipleOwnershipAllWorkersCheckCorrectRPCs2"), FWorkerDefinition::AllWorkers, PossessesCorrectPawns2, nullptr,
+		[this](float DeltaTime) {
+			RequireEqual_Int(1, MultipleOwnershipPawns[0]->ReceivedRPCs, TEXT("Pawn 0 received correct number of RPCs"));
+			RequireEqual_Int(0, MultipleOwnershipPawns[1]->ReceivedRPCs, TEXT("Pawn 1 received correct number of RPCs"));
+			FinishStep();
+		},
+		10.0f);
+
+	// The server makes Client 1's PlayerController possess the second MultipleOwnershipPawn.
+	AddStep(TEXT("SpatialTestMultipleOwnershipPossessPawn2"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		APlayerController* PlayerController =
+			Cast<APlayerController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+		PlayerController->Possess(MultipleOwnershipPawns[1]);
+		MultipleOwnershipPawns[0]->SetOwner(PlayerController);
+
+		FinishStep();
+	});
+
+	// Client 1 sends an RPC from both MultipleOwnershipPawns
+	AddStepFromDefinition(ClientSendRPCStepDefinition, FWorkerDefinition::Client(1));
+
+	auto PossessesCorrectPawns3 = [this]() {
+		APlayerController* PlayerController =
+			Cast<APlayerController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+		return (PlayerController == nullptr)
```

Comment:
hm, actually why do we return true if the playerController is null?
Seems like that should never happen

Reply:
ah, nevermind, the other client does not see the playerController of the first client

Reply:
The test gets run on both clients and the server, and on client 2 we can't access that player controller, so it returns null. Also, looking at it now it doesn't seem to have the possession checks which is really weird.